### ISSUE TITLE
RCON

### ISSUE THREAD

**Issue by FarrisBris** [lucko/LuckPerms#1365]
When I execute Luckperms commands thru RCON it returns a blank line.

**Comment by Chappy202** [comment]
What version of luckperms are you running?

**Comment by FarrisBris** [comment]
4.3.2

**Comment by Chappy202** [comment]
Try upgrading to 4.3.17 and see if it fixes your issue

**Comment by lucko** [comment]
It's an issue with plugins which handle commands async / attempt to send responses to command senders at a later time.

**Comment by FarrisBris** [comment]
> 
> 
> It's an issue with plugins which handle commands async / attempt to send responses to command senders at a later time.

As of my understanding, there is no way at all that RCON will work with LuckPerms?

**Comment by lucko** [comment]
At the moment, yes that is the case.

**Comment by FarrisBris** [comment]
> 
> 
> At the moment, yes that is the case.
> 
> If Spigot merged the previously mentioned pull request, then it would work, but I can't influence that process.

Well that is a huge rip. Thank you.

**Comment by lucko** [comment]
Yep indeed it is a shame. :/

**Comment by brandinarsenault** [comment]
@lucko,

Sorry to comment on this old issue thread. I'm assuming this pull request has not been merged and there is no update posted? Unfortunately, I cannot review the pull request using that link as my login for their Jira does not grant me access.

Thank you for all your work on this, it is a fantastic plugin and project!

Brandin.

**Comment by lucko** [comment]
Yep, pull request is still open, but hasn't received any comment / update in over a year.

**Comment by straight-shoota** [comment]
This is really a bummer. I suppose the fix hasn't landed yet still.

I'd like to issue lp commands over RCON to initialize groups and memberships from a web UI. The commands itself execute, so that's technically fine there's no feedback and I can't verify the existing settings.
Are there any suggestions how to achieve that instead? I could directly access the database, but I figured letting LuckPerms handle that would avoid interference and potential consistency issues. But looks like this might be the better solution when RCON doesn't work properly.

**Comment by maximov-ru** [comment]
Pleease reopen this bug.

**Comment by brandinarsenault** [comment]
> Pleease reopen this bug.

This isn't a LuckPerms bug - so theres no reason to have this open. It's on Spigot to fix, not Lucko.

**Comment by macbrayne** [comment]
Weirdly enough the Fabric version doesn't work either with RCON, this shouldn't be due to some Spigot limitations, right?

**Comment by HyperPaint** [comment]
I confirm. When working with Fabric on latest version, i reproduces the same bug. RCON isn't working

**Comment by BNTFryingPan** [comment]
a new issue for rcon on fabric is more likely to be noticed than just a comment on a *closed* issue

**Comment by GamersOr** [comment]
Can someone help me? I have the same bug like that the rcon can excute "say", but if I do a luckperms then its not working

**Comment by HyperPaint** [comment]
> Can someone help me? I have the same bug like that the rcon can excute "say", but if I do a luckperms then its not working

You can use external storage like MySQL and configure luckperms to reload config from storage once every few minutes. Then, you can modify the rights on the fly, but through SQL. This will require additional skills

**Comment by underscore11code** [comment]
Let's not continue to post on a 5 year old issue please. Feel free to open a fresh issue, though it'll most likely continue to be blocked by the aforementioned Spigot PR on the Bukkit platform.

@HyperPaint Note that hand modifying our SQL database storage methods is unsupported, if you break the database in the process of manually editing, you're on your own for either repairing it or reloading a backup.

**Comment by Gioxaa** [comment]
so there is no solution guys?
